### ISSUE TITLE
Support `data:` URLs in `fetch`

### ISSUE THREAD

**Issue by ElYaiko** [oven-sh/bun#3956]
### What version of Bun is running?

0.7.2

### What platform is your computer?

Darwin 22.5.0 arm64 arm

### What steps can reproduce the bug?

Using this method to convert an image from Base64 to Blob:

```
var url = "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU5ErkJggg=="

fetch(url)
.then(res => res.blob())
.then(console.log)
```

### What is the expected behavior?

Convert Base64 string to Blob.

### What do you see instead?

FailedToOpenSocket: Was there a typo in the url or port?

### Additional information

This works well on NodeJS.

**Comment by Jarred-Sumner** [comment]
`data:` URI is not supported yet

not for any good reason tbh

**Comment by Jarred-Sumner** [comment]
Fixed by @dylan-conway in #4000
